go/signedexchange/signingalgorithm.go

```diff
@@ -0,0 +1,69 @@
+package signedexchange
+
+import (
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rsa"
+	"encoding/asn1"
+	"fmt"
+	"io"
+	"math/big"
+)
+
+type signer interface {
+	sign(m []byte) ([]byte, error)
+}
+
+type rsaPSSSigner struct {
+	privKey *rsa.PrivateKey
+	hash    crypto.Hash
+	rand    io.Reader
+}
+
+func (s *rsaPSSSigner) sign(m []byte) ([]byte, error) {
+	hash := s.hash.New()
+	hash.Write(m)
+	return rsa.SignPSS(s.rand, s.privKey, s.hash, hash.Sum(nil), nil)
```

Comment:
@nyaxt pointed out that the last `nil` here is a bug: RSA-PSS requires that the salt length equals the hash length, so you need `&PSSOptions{SaltLength:rsa.PSSSaltLengthEqualsHash}`.

Reply:
I'm preparing a separate PR to address this. Will follow up shortly.

Reply:
https://github.com/hajimehoshi/webpackage/pull/2

Reply:
Done.